src/main/kotlin/cache/loaders/TextureLoader.kt

```diff
@@ -49,35 +56,95 @@ class TextureLoader(cacheLibrary: CacheLibrary) {
+                return null
             }
+            
+            def.field1777 = inputStream.readUnsignedShort()
+            def.field1778 = inputStream.get().toInt() != 0
+            def.id = id
+            
+            if (inputStream.remaining() < 1) {
+                return null
+            }
+            
+            val count: Int = inputStream.readUnsignedByte()
+            
+            // Validate count is reasonable
+            if (count < 0 || count > 255) {
+                return null
+            }
+            
+            val files = IntArray(count)
+            
+            // Check if we have enough bytes for all file IDs
+            if (inputStream.remaining() < count * 2) {
+                return null
+            }
+            
+            for (i in 0 until count) {
+                files[i] = inputStream.readUnsignedShort()
+            }
+            def.fileIds = files
+            
+            if (count > 1) {
+                // Check if we have enough bytes for field1780
+                if (inputStream.remaining() < count - 1) {
+                    return null
+                }
+                def.field1780 = IntArray(count - 1)
+                for (var3 in 0 until count - 1) {
+                    def.field1780[var3] = inputStream.readUnsignedByte()
+                }
+            }
+            
+            if (count > 1) {
+                // Check if we have enough bytes for field1781
+                if (inputStream.remaining() < count - 1) {
+                    return null
+                }
+                def.field1781 = IntArray(count - 1)
+                for (var3 in 0 until count - 1) {
+                    def.field1781[var3] = inputStream.readUnsignedByte()
+                }
+            }
+            
+            // Check if we have enough bytes for field1786 (count * 4 bytes for ints)
+            if (inputStream.remaining() < count * 4) {
+                return null
+            }
+            def.field1786 = IntArray(count)
+            for (var3 in 0 until count) {
+                def.field1786[var3] = inputStream.int
+            }
+            
+            // Check if we have enough bytes for the last two fields
+            if (inputStream.remaining() < 2) {
+                return null
+            }
+            
+            def.field1783 = inputStream.readUnsignedByte()
+            def.field1782 = inputStream.readUnsignedByte()
+            return def
+        } catch (e: java.nio.BufferUnderflowException) {
+            // Texture data is incomplete or corrupted, skip this texture
+            System.err.println("Warning: Texture $id has incomplete or corrupted data, skipping.")
+            return null
+        } catch (e: Exception) {
+            // Any other error loading the texture, skip it
+            System.err.println("Warning: Failed to load texture $id: ${e.message}, skipping.")
+            return null
         }
```

Comment:
a lot of manual checks for remaining buffer + catching in load function. failure is an exceptional condition, it should be fine to let it bubble up and get handled by the caller